docs/configuring-a-resource.md

```diff
@@ -0,0 +1,410 @@
+  attribute.secret: ABCxyzRedacted==
+  attribute.ses_smtp_password_v4: QQ00REDACTED==
+  aws_access_key_id: QUtJQVk0QUZUVFNFNDI2TlhKS0I=
+  aws_secret_access_key: ABCxyzRedacted==
+kind: Secret
+```
+
+### Late Initialization Behavior
+Terrajet runtime automatically performs late-initialization during
+an [`external.Observe`] call with means of runtime reflection.
+State of the world observed by Terraform CLI is used to initialize
+any `nil`-valued pointer parameters in the managed resource's `spec`.
+In most of the cases no custom configuration should be necessary for
+late-initialization to work. However, there are certain cases where
+you will want/need to customize late-initialization behaviour. Thus,
+Terrajet provides an extensible [late-initialization customization API]
+that controls late-initialization behaviour.
+
+The associated resource struct is defined [here](https://github.com/crossplane-contrib/terrajet/blob/c9e21387298d8ed59fcd71c7f753ec401a3383a5/pkg/config/resource.go#L91) as follows:
+```go
+// LateInitializer represents configurations that control
+// late-initialization behaviour
+type LateInitializer struct {
+    // IgnoredFields are the canonical field names to be skipped during
+    // late-initialization
+    IgnoredFields []string
+}
+```
+Currently, it only involves a configuration option to specify
+certain `spec` parameters to be ignored during late-initialization.
+Each element of the `LateInitializer.IgnoredFields` slice represents
+the canonical path relative to the parameters struct for the managed resource's `Spec`
+using `Go` type names as path elements. As an example, with the following type definitions:
+```go
+type Subnet struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    Spec              SubnetSpec   `json:"spec"`
+    Status            SubnetStatus `json:"status,omitempty"`
+}
+
+type SubnetSpec struct {
+    ForProvider     SubnetParameters `json:"forProvider"`
+    ...
+}
+
+type DelegationParameters struct {
+    // +kubebuilder:validation:Required
+    Name *string `json:"name" tf:"name,omitempty"`
+    ...
+}
+
+type SubnetParameters struct {
+    // +kubebuilder:validation:Optional
+    AddressPrefix *string `json:"addressPrefix,omitempty" tf:"address_prefix,omitempty"`
+    // +kubebuilder:validation:Optional
+    Delegation []DelegationParameters `json:"delegation,omitempty" tf:"delegation,omitempty"`
+    ...
+}
+```
+If you would like to have the late-initialization library *not* to process the
+[`address_prefix`] parameter field, then the following configuration where we
+specify the parameter field path is sufficient:
+
+```go
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("azurerm_subnet", func(r *config.Resource) {
+		r.LateInitializer = config.LateInitializer{
+			IgnoredFields: []string{"address_prefix"},
+		}
+	})
+}
+```
+
+In most cases, custom late-initialization configuration will not be necessary. 
```

Comment:
Up until this point users still don't know why they might need to customize this behavior. So, I think having an example case at the beginning like `if you see that behavior, then you need to customize this by following these steps`.

Reply:
Crosslinking to #153 
//cc @ulucinar